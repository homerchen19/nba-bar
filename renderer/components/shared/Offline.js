import React, { PureComponent } from 'react';
import styled from 'styled-components';
import { Flex } from 'antd-mobile';

import { colors } from '@styles/theme';
import Layout from './Layout';

const Content = styled(Flex)`
  flex: 1;
  width: 100%;
  height: 100%;
  padding: 0.5rem;
`;

const Oops = styled.h1`
  color: ${colors.red};
  font-size: 1.2rem;
  font-weight: 500;
  text-align: center;
`;

export default class Offline extends PureComponent {
  render() {
    return (
      <Layout>
        <Content direction="column" justify="center">
          <Oops>
            Oops, offline.
            <br />
            This App requires the internent to be connected ðŸ“¡
          </Oops>
        </Content>
      </Layout>
    );
  }
}
