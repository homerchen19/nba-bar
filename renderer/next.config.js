const withCSS = require('@zeit/next-css');
const withImages = require('next-images');

module.exports = withImages(
  withCSS({
    cssModules: true,
    webpack(config, { isServer }) {
      config.target = 'electron-renderer';

      if (isServer) {
        const antStyles = /antd-mobile\/.*?\/style.*?/;
        const origExternals = [...config.externals];
        config.externals = [
          (context, request, callback) => {
            if (request.match(antStyles)) return callback();
            if (typeof origExternals[0] === 'function') {
              origExternals[0](context, request, callback);
            } else {
              callback();
            }
          },
          ...(typeof origExternals[0] === 'function' ? [] : origExternals),
        ];

        config.module.rules.unshift({
          test: antStyles,
          use: 'null-loader',
        });
      }

      return config;
    },
    exportPathMap() {
      return {
        '/home': { page: '/home' },
        '/live': { page: '/live' },
        '/preview': { page: '/preview' },
        '/scoreboard': { page: '/scoreboard' },
        '/settings': { page: '/settings' },
        '/standings': { page: '/standings' },
      };
    },
  })
);
