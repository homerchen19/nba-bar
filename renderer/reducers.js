import { combineReducers } from 'redux';

import homeReducer from './containers/Home/reducer';
import previewReducer from './containers/Preview/reducer';
import liveReducer from './containers/Live/reducer';
import scoreboardReducer from './containers/Scoreboard/reducer';
import standingsReducer from './containers/Standings/reducer';
import settingsReducer from './containers/Settings/reducer';

const rootReducer = combineReducers({
  home: homeReducer,
  preview: previewReducer,
  live: liveReducer,
  scoreboard: scoreboardReducer,
  standings: standingsReducer,
  settings: settingsReducer,
});

export default rootReducer;
