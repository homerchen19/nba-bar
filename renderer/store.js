import { createStore, applyMiddleware, compose } from 'redux';
import thunk from 'redux-thunk';
import logger from 'redux-logger';

import rootReducer from './reducers';

const middleware = [thunk, logger];

const enhancers = compose(
  applyMiddleware(...middleware),
  typeof window !== 'undefined' &&
    window.devToolsExtension &&
    process.env.NODE_ENV !== 'production'
    ? window.devToolsExtension()
    : f => f
);

export default function configureStore(initialState = {}) {
  const store = createStore(rootReducer, initialState, enhancers);

  return store;
}
