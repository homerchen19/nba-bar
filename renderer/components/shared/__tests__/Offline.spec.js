import React from 'react';
import renderer from 'react-test-renderer';

import Offline from '../Offline';

it('be defined', () => {
  expect(Offline).toBeDefined();
});

it('match snapshot', () => {
  const tree = renderer.create(<Offline />).toJSON();

  expect(tree).toMatchSnapshot();
});
