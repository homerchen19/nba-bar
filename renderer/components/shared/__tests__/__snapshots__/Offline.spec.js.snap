// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className="Layout-sc-1qlvb6x-0 iYCmAx"
>
  <div
    className="am-flexbox Offline__Content-sc-1hops6p-0 YXDsW am-flexbox-dir-column am-flexbox-justify-center am-flexbox-align-center"
  >
    <h1
      className="Offline__Oops-sc-1hops6p-1 jEifYf"
    >
      Oops, offline.
      <br />
      This App requires the internent to be connected ðŸ“¡
    </h1>
  </div>
</div>
`;
